enigma-core/app/src/networking/ipc_listener.rs

```diff
@@ -281,6 +281,8 @@ pub(self) mod handling {
             let delta = IpcStatusResult { address, key, status };
             errors.push(delta);
         }
+        // since a new delta was added the state is no longer updated
+        db.update_state_status(false);
```

Comment:
I didn't get into details, so maybe this comment is wrong, but shouldn't the same be done for `update_new_contract_on_deployment`?

Reply:
at first, I thought we won't need it, however, now that I'm thinking of it, you're right. I'll add this as well.

Reply:
We don't need it, since every contract comes with a delta as well.

Reply:
you're right! we do need this. I added it.